### ISSUE TITLE
Error when trying to install fabric in a different location than the default.

### ISSUE THREAD

**Issue by calloatti** [FabricMC/fabric-installer#16]
It seems the fabric installer expects the install location specified to already exist, so instead of creating the directory for the install it raises an error.

![image](https://user-images.githubusercontent.com/12817862/58379301-064ca480-7f78-11e9-9e6a-df15b3d73e7c.png)

The installer should allow the user to specify a different location than the default, just as the vanilla launcher allows it when setting up a new profile.

See https://help.mojang.com/customer/en/portal/articles/731321-multiple-accounts-on-one-computer

**Comment by modmuss50** [comment]
I could create the directory, but honestly I should add a check to ensure that the launcher is in-fact already present in that directory.

**Comment by calloatti** [comment]
Same thing with the server install. It fails if the specified folder does not exist:

![image](https://user-images.githubusercontent.com/12817862/58516737-6a8d8500-817f-11e9-850a-ab95755ad7f6.png)

**Comment by calloatti** [comment]
About the client install, and the "no launcher directory found", there are other problems with the Fabric Installer beyond that error, that I will try to explain.

TL;DR: Instead of trying to fix this, advanced users should just use multimc imho.

The vanilla launcher defaults to `%appdata%\.minecraft` as the folder where it saves `launcher_profiles.json` and other folders with common data. 

When you create a new instance, by default that instance will be "installed" in the default `%appdata%\.minecraft`, so saved games will be shared among all instances under the `%appdata%\.minecraft` default folder.

A user may pick a different folder to install an instance isolated from the default folder, as explained in:

https://help.mojang.com/customer/en/portal/articles/731321-multiple-accounts-on-one-computer

When an instance is created in a different folder, the instance data is still saved inside `%appdata%\.minecraft\launcher_profiles.json`

For example I create an instance and set the game directory to `%appdata%\.testvanilla` and run the game,  create a world, save it, exit the game.

Content of %appdata%\.testvanilla:
```
logs
realms_persistence.json
resourcepacks
saves
usercache.json
```
Content of %appdata%\.minecraft
```
assets
launcher_log.txt
launcher_profiles.json
libraries
nativelog.txt
versions
```
The instance data in `%appdata%\.minecraft\launcher_profiles.json`:

```
    "7cc7819f408a4ade5ca87f06101eb64f": {
      "name": "testvanilla",
      "type": "custom",
      "created": "2019-05-28T22:39:53.136Z",
      "lastUsed": "2019-05-28T23:24:58.123Z",
      "lastVersionId": "1.14.2",
      "gameDir": "C:\\Users\\username\\AppData\\Roaming\\.testvanilla"
    }

```
Now I create a folder` %appdata%\testfabric` to get around the "no launcher directory found" error and create an instance using the fabric installer, using that folder. I check "Create profile" and click install. The install finishes and displays "Done".

Now I open the MC launcher and the instance I just created using the fabric installer is not listed at all.

### ISSUE 1

The fabric installer could not find the file `launcher_profiles.json` to add the new instance data, so it silently fails. 

It could not find the file because it assumes the folder `%appdata%\.testfabric` is the main folder for the vanilla launcher, when the main folder is really `%appdata%\.minecraft`. It should edit the `launcher_profiles.json` file located there.

Question: How does the installer then find the location of the Vanilla MC data folder? 
Answer: I don't know. It may be safe to assume the folder will always be` %appdata%\.minecraft`, even though there seems to be ways to change it that I have not tested.

### ISSUE 2:

Since the launcher assumes that the folder specified by the user is the main MC data folder, it creates the fabric jar and json here:

` %appdata%\.testfabric\versions\fabric-loader-0.4.8+build.154-1.14.2+build.2` 

 Instead of correctly creating those files here:
 
` %appdata%\.minecraft\versions\fabric-loader-0.4.8+build.154-1.14.2+build.2` 

### Manual fix:
 
Move the `appdata%\.testfabric\versions\fabric-loader-0.4.8+build.154-1.14.2+build.2` folder to `%appdata%\.minecraft\versions`

Edit `%appdata%\.minecraft\launcher_profiles.json` and edit the testvanilla profile to point to the fabric instance:

```
    "7cc7819f408a4ade5ca87f06101eaaaa": {
      "name": "testfabric",
      "type": "custom",
      "created": "2019-05-28T22:39:53.136Z",
      "lastUsed": "2019-05-28T23:24:58.123Z",
      "lastVersionId": "fabric-loader-0.4.8+build.154-1.14.2+build.2",
      "gameDir": "C:\\Users\\username\\AppData\\Roaming\\.testfabric"
    }

```
Now the instance shows in the list, and launches MC succesfully.

I hope this novel I wrote helps.

**Comment by calloatti** [comment]
Basically, I did not know how to use the installer.
